libkbfs/node_cache.go

```diff
@@ -146,8 +161,16 @@ func (ncs *nodeCacheStandard) Get(ref BlockRef) Node {
 		panic(InvalidBlockRefError{ref})
 	}
 
+	var rootWrappers []func(Node) Node
+	defer func() {
+		if n != nil {
+			n = ncs.wrapNodeStandard(n, rootWrappers)
+		}
+	}()
```

Comment:
`rootWrappers` is always assigned with `ncs.rootWrappers`, so it doesn't seem necessary. Is there a reason not to just use `ncs.rootWrappers` in the deferred function?

If the concern is that `ncs.rootWrappers` may change, I think the deferred function should still be called under the lock because even though it's assigned to a different variable, it still shares the same underlying buffer with all other references. But it doesn't seem like `ncs.rootWrappers` can change.

(same below)

Reply:
There's no guarantee `ncs.rootWrappers` won't change, so I was trying to save it while under lock.  But you're right, those places should use a copy.  Will fix.

Reply:
Ah I missed the `AddRootWrapper ` method. As long as the deferred function is called under the lock (i.e. defer it after the deferred unlock) and it'd be fine right?

Reply:
Yes, but the whole point here is to apply the wrappers while *not* under lock, because that leads to deadlock.

Reply:
Sounds good. Sorry for going back and forth and causing the confusion!